### ISSUE TITLE
Refactor rename code to use extended `TModel` instead of `WorkspaceInfo`

### ISSUE THREAD

**Issue by toinehartman** [usethesource/rascal-language-servers#446]
Currently, these is a lot of overlap between the `WorkspaceInfo` and `TModel` constructors. Instead of the `WorkspaceInfo`, we could use an augmented `TModel`.

**Comment by DavyLandman** [comment]
@toinehartman I think this has happened?

**Comment by toinehartman** [comment]
Correct! https://github.com/usethesource/rascal-language-servers/blob/644888ee7ec1b15f4481d58fd675ea384e04d1ce/rascal-lsp/src/main/rascal/lang/rascal/lsp/refactor/WorkspaceInfo.rsc#L77-L81
